### DISCUSSION TITLE
Make non-existing favicons render consistently

### DISCUSSION THREAD

**Discussion by RobertJoonas** [plausible/analytics#2088] [Anything Else]
We are currently fetching all favicons from DuckDuckGo. Usually, if a favicon or domain does not actually exist, they return the default gray circle image with an arrow. Example: https://icons.duckduckgo.com/ip3/non-existing-domain-xxx.com.ico.

With some domains though, they return a 404, not the default image. Examples:

https://icons.duckduckgo.com/ip3/email-non-existing-domain-xxx.com.ico.

In that case, browser displays a broken image icon. Would be nice to make it consistent and render the same gray circle image in case of a 404 too.

**Comment by GarveScott-Lodge** [comment]
I notice that the dashboard seems to be showing favicons which DuckDuckGo doesn't. Not sure how that's working, but thanks guys!

**Reply by metmarkosaric** [reply]
thanks @GarveScott-Lodge! we made a tweak to this recently. details here https://github.com/plausible/analytics/pull/2295
